File: 프로그래머스/0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.cpp
#include <string>
#include <iostream>
#include <vector>

using namespace std;

int getGcd(int a, int b) 
{
   if (b == 0) 
       return a;
    return getGcd(b, a % b);
}

int solution(int n) 
{
    int gcd = getGcd(6, n);
    int lcm = (6 / gcd) * (n / gcd) * gcd;
    
    return lcm / 6;
}

File: 프로그래머스/0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.js
function fnGcd(a, b)
{
    if (b === 0)
        return a
    return fnGcd(b, a % b);
}

function solution(n) {
    let answer = 0;
    let gcd = fnGcd(6, n);
    
    answer = (n / gcd) * (6 / gcd) * gcd;
    
    return answer / 6;
}